pkgs/stdenv/generic/setup.sh

```diff
@@ -1121,7 +1121,11 @@ substituteAllInPlace() {
 # the environment used for building.
 dumpVars() {
     if [ "${noDumpEnvVars:-0}" != 1 ]; then
-        install -m 0600 <(export 2>/dev/null) "$NIX_BUILD_TOP/env-vars" || true
+        # On darwin, install(1) cannot be called with /dev/stdin or fd from process substitution
+        # so first we create the file and then write to it
+        # See https://github.com/NixOS/nixpkgs/issues/335016
+        install -m 0600 /dev/null "$NIX_BUILD_TOP/env-vars" || true
+        export 2>/dev/null >| "$NIX_BUILD_TOP/env-vars" || true
     fi
```

Comment:
I believe that the idea of #331576 is *not* to write the `env-vars` file if it doesn't have the right permissions.

But the current sequence of commands in this PR means that if the `install` fails for some reason, then the `export` will still write to the file.

I think this could be better:

``` sh
{
    touch $NIX_BUILD_TOP/env-vars" &&
    chmod 600 $NIX_BUILD_TOP/env-vars" &&
    export 2>/dev/null >| "$NIX_BUILD_TOP/env-vars"
} || true
```

while also being more clear about the intent.

Reply:
Thanks, yeah that was indeed why I attempted to do it in one pass initially.